src/rebar3_hex_file.erl

```diff
@@ -0,0 +1,35 @@
+-module(rebar3_hex_file).
+
+-export([
+         expand_paths/2,
+         update_app_src/2
+        ]).
+
+expand_paths(Paths, Dir) ->
+    AbsDir = filename:absname(Dir),
+    Files = lists:flatmap(fun dir_files1/1, [filename:join(Dir, P) || P <- Paths]),
+    [{relative_path(F1, AbsDir), F1} || F1 <- filter_regular(Files)].
```

Comment:
@ferd Pointing you to this since the PR you commented on is actually the bulk of this one until it gets merged. It appears rebar_dir already provides us with a tried and true approach for making a relative path.